src/test/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdaoUnitTest.java

```diff
@@ -0,0 +1,123 @@
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+class AzureBlobStorePdaoUnitTest {
+  @Mock private ProfileDao profileDao;
+  @Mock private AzureContainerPdao azureContainerPdao;
+  @Mock private AzureResourceConfiguration azureResourceConfiguration;
+  @Mock private AzureResourceDao azureResourceDao;
+  @Mock private AzureAuthService azureAuthService;
+  @Mock private AzureBlobService azureBlobService;
+  @Mock private FileIdService fileIdService;
+  @Mock private GcsProjectFactory gcsProjectFactory;
+  @Mock private GcsPdao gcsPdao;
+
+  private static final AuthenticatedUserRequest TEST_USER =
+      AuthenticationFixtures.randomUserRequest();
+
+  private static final UUID BILLING_PROFILE_ID = UUID.randomUUID();
+  private static final UUID SUBSCRIPTION_ID = UUID.randomUUID();
+  private static final String RESOURCE_GROUP_NAME = "resourceGroupName";
+  private static final String APPLICATION_DEPLOYMENT_ID =
+      String.format("subscriptions/%s/resourceGroups/%s", SUBSCRIPTION_ID, RESOURCE_GROUP_NAME);
+  private static final String SIGNED_URL =
+      "https://signedUrl.blob.core.windows.net/uuid/blobPath?sig=SIGNATURE";
+
+  private BillingProfileModel billingProfileModel;
+  private AzureApplicationDeploymentResource applicationDeploymentResource;
+  private AzureStorageAccountResource storageAccountResource;
+  private AzureBlobStorePdao azureBlobStorePdao;
+
+  @BeforeEach
+  void setup() {
+    billingProfileModel = new BillingProfileModel().id(BILLING_PROFILE_ID);
+    azureBlobStorePdao =
+        new AzureBlobStorePdao(
+            profileDao,
+            azureContainerPdao,
+            azureResourceConfiguration,
+            azureResourceDao,
+            azureAuthService,
+            azureBlobService,
+            fileIdService,
+            gcsProjectFactory,
+            gcsPdao);
+
+    applicationDeploymentResource =
+        new AzureApplicationDeploymentResource()
+            .azureApplicationDeploymentId(APPLICATION_DEPLOYMENT_ID)
+            .azureResourceGroupName(RESOURCE_GROUP_NAME);
+    storageAccountResource =
+        new AzureStorageAccountResource()
+            .applicationResource(applicationDeploymentResource)
+            .name("storageAccountResourceName")
+            .profileId(BILLING_PROFILE_ID);
+  }
+
+  void mocksForDeleteBlobParquet() {
+    when(profileDao.getBillingProfileById(BILLING_PROFILE_ID)).thenReturn(billingProfileModel);
+    when(azureContainerPdao.getDestinationContainerSignedUrl(any(), any(), any()))
+        .thenReturn(SIGNED_URL);
+
+    BlobCrl blobCrl = mock(BlobCrl.class);
+    when(azureBlobService.getBlobCrl(any())).thenReturn(blobCrl);
+    when(blobCrl.deleteBlobsWithPrefix(any())).thenReturn(true);
```

Comment:
Quoting my other comment about the preexisting bug in BlobUrl construction with added emphasis:

> As long as the new unit tests verify the expected behavior -- **that we are deleting blobs with the expected prefix** -- then I'd be comfortable with filing a separate ticket to fix this latent bug, since it doesn't presently impact users.

Presently, this stub is overly permissive.  Rather than returning `true` on `any()` prefix, can we pass in the specific prefixes we expect for our test cases?  Otherwise I don't take much stock in the meaning of a `true` result.